### DISCUSSION TITLE
API Rules with @request.data

### DISCUSSION THREAD

**Discussion by bijsven** [pocketbase/pocketbase#4805] [Q&A] [answered]
👋 I am trying to get this to work, but for some reason it just wont work for me:

List API Rule:
`@request.data.DeviceToken:isset = true`

API request:
```await pb.collection('auth').getFirstListItem(`DeviceToken = "${DeviceToken}"`)```

When removing the hole API rule, it does work, but when adding (above api rule) or
`@request.data.DeviceToken = DeviceToken` it does not work.

Did I overlook something?

Thanks,

**Comment by ganigeorgiev** [comment, marked as answer]
`@request.data.*` field targets submitted **body fields** (_with the refactoring it will be renamed to `@request.body.*` to avoid the confusion_). You could find information about the available `@request.*` fields in https://pocketbase.io/docs/api-rules-and-filters/#filters-syntax.

What you are submitting is a `filter` **query parameter** with a value `DeviceToken = yourDeviceToken`.
Or in other words, if you inspect the request in the browser you'll see something like:
```
GET /api/collections/auth/records?page=1&perPage=1&skipTotal=1&filter=DeviceToken%20%3D%20yourDeviceToken
```

It is also not clear what is the purpose of the API rule.

If the goal is to allow only the list of records that have a "DeviceToken" field matching the one from your request, then you can use a filter rule like:

```
DeviceToken = @request.query.token
```

And submit your request with:

```js
const token = ...
// submit an empty filter and the token as its own query parameter
await pb.collection('auth').getFirstListItem("", { token })
```

If the above doesn't help, please elaborate a little more on your use case (what is the purpose of the `DeviceToken`?) and export your collections configuration as json (_Admin UI > Settings > Export collections_).

**Reply by bijsven** [reply]
Thank you! - That worked!
